rclpy/rclpy/__init__.py

```diff
@@ -70,7 +74,9 @@ def init(*, args: List[str] = None, context: Context = None) -> None:
     context = get_default_context() if context is None else context
     # imported locally to avoid loading extensions on module import
     from rclpy.impl.implementation_singleton import rclpy_implementation
-    return rclpy_implementation.rclpy_init(args if args is not None else sys.argv, context.handle)
+    global NodeNameNonExistentError
+    rclpy_implementation.rclpy_init(args if args is not None else sys.argv, context.handle)
+    NodeNameNonExistentError = rclpy_implementation.NodeNameNonExistentError
```

Comment:
Why should this be the exposed C symbol? Wouldn't an exception defined in Python make more sense here?

Reply:
I'll weight in here, because I had a similar dilemma while figuring out how to raise appropriate exceptions upon failed argument parsing in `rclpy`: either defining an exception on Python and importing it from C or defining it in C and importing it on Python. And the latter seems more natural to me too. 

But then this. Why is that C extensions cannot be imported at module level?

Reply:
A little of history, see: https://github.com/ros2/rclpy/pull/147

I would rather not limit C extensions to be loaded after the module. Instead, it should be possible to import them at module level too.
I have a solution in mind, that makes possible to import C extensions at module level, and shows a clearer relative path when testing. I will try if it works or not.

Reply:
My current solution has a big problem, the exception is only valid after `rclpy_init`. I will switch to an exception defined in Python if I don't successfully avoid the limitation of importing C extensions at module level.